components/google_sheets/sources/common/http-based/sheet.mjs

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable pipedream/required-properties-key, pipedream/required-properties-name,
+  pipedream/required-properties-version, pipedream/required-properties-description,
+  pipedream/required-properties-type */
+
+import { v4 as uuid } from "uuid";
+
+import base from "./base.mjs";
+
+/**
+ * This source watches for changes to a specific spreadsheet in the user's Google Drive.
+ */
+export default {
+  ...base,
+  hooks: {
+    ...base.hooks,
+    /**
+     * Called when a component is created or updated. Handles all the logic
+     * for starting and stopping watch notifications tied to the desired file.
+     */
+    async activate() {
+      const channelID = this._getChannelID() || uuid();
+
+      const {
+        startPageToken,
+        expiration,
+        resourceId,
+      } = await this.googleSheets.activateFileHook(
+        channelID,
+        this.http.endpoint,
+        this.getSheetId(),
+      );
+
+      // Save metadata on the subscription so we can stop / renew later
+      // Subscriptions are tied to Google's resourceID, "an opaque value that
+      // identifies the watched resource". This value is included in request headers
+      this._setSubscription({
+        resourceId,
+        expiration,
+      });
+      this._setChannelID(channelID);
+      if (startPageToken) {
+        this._setPageToken(startPageToken);
+      }
+
+      await this.takeSheetSnapshot();
+    },
+  },
```

Comment:
If you move the `activate()` hook to base.mjs, this hooks section can be removed.
```suggestion
```